docs/entities/registeredservice.md

```diff
@@ -0,0 +1,45 @@
+a human or a program. This flexibility allows for the service provisioning and service discovery operations
+to be decoupled from representation of that service in a service catalog. The service catalog can be consumed by
+Kubernetes applications that are unaware of the lifecycle of the service. The creation of a registered server signals
+and update to the service catalog can be be advertised to the application developers. A RegisteredService can be
+claimed to be used by a specific application. If the application does not require the service any longer, it can
+remove the claim and the registered service is put back in the pool of available services.
+
+## Specification
+
+The definition of a RegisteredService can be obtained directly from our [RegisteredService CRD](../../config/crd/bases/primaza.io_registeredservices.yaml). The specification contains 2 required properties and
+3 optional properties:
+
+- ServiceClassIdentity: A set of key/value pairs that identify the service class. Examples of service class identity keys include type of service, and provider of service. This property is required.
+- ServiceEndpointDefinition: A set of key/value pairs that provides two pieces of information: the service connectivity and
+the service authentication. Values can be either a string or a reference to a secret field. This property is required.
+- Constraints: Restrictions of the registered service. A constrain could be a specific environment where the
+service cannot be claimed from. As an example, a constrain could describe that I don't want this service to be used on any environment except for production. This property is optional, when it is absent, it means there is not constrains.
+- HealthCheck: A mechanism to be able to verify the service is online and ready to use. One way this can be accomplished is
+by providing an image containing a client that can be run to test connectivity and authentication. This property is optional, when it is absent, it means the service will be considered available as soon as it is registered. 
+- SLA: Provides multiple levels of resiliency, scalability, fault tolerance and security. This allows claims to take into account the robustness of service. This property is optional, when it is absent, it means that there is no distinctions between services given the SLA.
+
+
+## Status
+
+The Status of the Service it is also defined under the [RegisteredService CRD](../../config/crd/bases/primaza.io_registeredservices.yaml). It contains only one required property to track state. The state could be one of the following: registered, available, claimed, unreachable. The registered service state is "registered" right after creation. If the health check is not defined or health check pass, the state would change to "available". Once a RegisteredService is claimed, its state moves to "claimed". On the other hand, if the health check comes back as a failure, the state would change to "unreachable".
```

Comment:
As far as I know, we also have `undetermined` as an available state.  We should call that out here and explain why it may be the state of a RegisteredService.

Reply:
I know from the doc we defined that. But looking closely, when would the state be undetermined?

Reply:
The way I understand it, `undetermined` can be useful when we don't actively know any of the other states.  It's a good way of indicating that something has gone wrong with the registered service, and that manual intervention may be required to resolve the issue.

As an example scenario for when this state would make sense, let's say a user defines a registered service with a health check that specifies a nonexistent container image.  The controller tries to run it, but it fails because the container image couldn't be pulled, resulting in a failure to run the health check.  In that case,  I think `undetermined` is the only state that makes sense.
 - `registered` to me means that primaza is aware of the resource, but doesn't know anything else about it.  It's effectively an acknowledgement that the resource exists.  However, in this case, we _do_ know more about it (namely, it has a faulty health check), so IMO `registered` doesn't convey the full context of what we've determined.  If we tried to run a health check that couldn't even run, we're well beyond the state of "we've acknowledged this resource exists".
 - `available` doesn't make sense to me here.  The user specified a health check, and so we should honor it by running it.  Since the health check couldn't determine availability, primaza shouldn't claim that we know otherwise.  IMO, primaza shouldn't hand out claims for services whose health check failed (regardless of _why_ it failed).
 - `claimed` makes no sense in this scenario, since there isn't a claim for the service.  We'd need for it to progress to `available` first, and with a failing health check, that's not happening.
 - `unreachable` doesn't quite feel right to me, since the problem here isn't with the service itself (it could be alive and well for all we know), but rather with the health check we have.  I'd prefer to reserve `unreachable` for "we ran the health check, and it told us the service isn't available"; in this scenario, we have "we tried to run the health check, but that failed", which is semantically different.

Reply:
An alternative idea is that we can stuff this underneath `unreachable`, but we provide more context to the user via a condition field in the status.  Either way, in my opinion we need tell the user that their service isn't available because of a problem in the registered service resource itself, not with the actual service.

Reply:
Thanks for the explanation @sadlerap I'm with you here, your example is perfect way to visualize it. Adding this changed